core/dbt/task/run.py

```diff
@@ -602,15 +602,15 @@ def _has_relation(self, model) -> bool:
         )
         return relation is not None
 
-    def _should_run_in_parallel(
-        self,
-        relation_exists: bool,
-    ) -> bool:
+    def should_run_in_parallel(self) -> bool:
         if not self.adapter.supports(Capability.MicrobatchConcurrency):
             run_in_parallel = False
-        elif not relation_exists:
+        elif not self.relation_exists:
             # If the relation doesn't exist, we can't run in parallel
             run_in_parallel = False
+        elif self.batch_idx == 0 or self.batch_idx == len(self.batches) - 1:
+            # First and last batch don't run in parallel
+            run_in_parallel = False
```

Comment:
This check could also be skipped if we're instead handling `force_sequential` to determine if we should even check `should_run_in_parallel` in `_submit_batch`. It'd be nice for this function to be less dependent on "where" it is, and I think this check breaks that.